Assets/Scripts/StandardSamples/Services/PlayerDataStorageService.cs

```diff
@@ -0,0 +1,468 @@
+/*
+* Copyright (c) 2021 PlayEveryWare
+* 
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+* 
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+* 
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+//#define ENABLE_DEBUG_PLAYERDATASTORAGE_SERVICE
+
+namespace PlayEveryWare.EpicOnlineServices.Samples
+{
+    using System;
+    using System.Diagnostics;
+    using System.Collections.Generic;
+    using System.Collections.Concurrent;
+
+    using UnityEngine;
+    using UnityEngine.Networking;
+
+    using Epic.OnlineServices;
+    using Epic.OnlineServices.Achievements;
+    using System.Threading.Tasks;
+    using Epic.OnlineServices.PlayerDataStorage;
+    using System.Threading;
+
+    /// <summary>
+    /// Class <c>AchievementsService</c> is a simplified wrapper for
+    /// EOS [Achievements Interface](https://dev.epicgames.com/docs/services/en-US/Interfaces/Achievements/index.html).
+    /// </summary>
+    public class PlayerDataStorageService : EOSService
+    {
+        private const uint MAX_CHUNK_SIZE = 4096;
+
+        /// <summary>
+        /// The EOS SDK can only handle one incoming or outgoing request at a time for several functions.
+        /// This semaphore can be used to gate calls, waiting until they're ready to be processed.
+        /// </summary>
+        private SemaphoreSlim _playerDataStorageSemaphore = new SemaphoreSlim(1);
```

Comment:
If _the EOS SDK can only handle one incoming or outgoing request at a time_, then this semaphore needs to be more widely accessible to other parts of the code. An example that comes immediately to mind is that you cannot have a TitleStorage request active at the same time as a PlayerStorage request - so _at least those two_ need a way to share this semaphore.